src/Microsoft.AspNet.Mvc.Core/Infrastructure/ObjectResultExecutor.cs

```diff
@@ -366,11 +369,19 @@ public virtual Task ExecuteAsync(ActionContext context, ObjectResult result)
                 throw new ArgumentNullException(nameof(acceptableContentTypes));
             }
 
-            var selectedFormatter = formatters.FirstOrDefault(
-                formatter => acceptableContentTypes.Any(
-                    contentType => formatter.CanWriteResult(formatterContext, contentType)));
+            foreach (var formatter in formatters)
+            {
+                foreach (var contentType in acceptableContentTypes)
```

Comment:
Thought you were going to make this the outer loop since users usually control the `[Produces]` list directly?